tests/test_preprocess.py

```diff
@@ -20,6 +28,163 @@
 ####################
 
 
+def test_whiten_bad_inputs():
+    # test window_size.ndim > 1 or step_size.ndim > 1
+    with pytest.raises(ValueError):
+        whiten(
+            img=np.ones((5, 5)),
+            window_size=np.ones((5, 5)),
+            step_size=np.array([3, 3]),
+            centered=False,
+            epsilon=1e-5,
+            type="PCA",
+        )
+    with pytest.raises(ValueError):
+        whiten(
+            img=np.ones((5, 5)),
+            window_size=np.array([3, 3]),
+            step_size=np.ones((5, 5)),
+            centered=False,
+            epsilon=1e-5,
+            type="PCA",
+        )
+
+    # test len(window_size) != len(step_size)
+    with pytest.raises(ValueError):
+        whiten(
+            img=np.ones((5, 5)),
+            window_size=np.array([3, 3]),
+            step_size=np.array([3, 3, 3]),
+            centered=False,
+            epsilon=1e-5,
+            type="PCA",
+        )
+
+    # test img.dnim != len(window_size)
+    with pytest.raises(ValueError):
+        whiten(
+            img=np.ones((5, 5)),
+            window_size=np.array([3, 3, 3]),
+            step_size=np.array([3, 3]),
```

Comment:
The error might be occurring because len(window_size) != len(step_size), I think if you make len(window_size) == len(step_size), then you can be sure that the error is because img.dnim != len(window_size)